agent/src/softwarecontaineragent.h

```diff
@@ -47,6 +48,66 @@ namespace softwarecontainer {
 
 static constexpr ContainerID INVALID_CONTAINER_ID = -1;
 
+class SoftwareContainerAgentError : public SoftwareContainerError
+{
+public:
+    SoftwareContainerAgentError():
+        m_message("SoftwareContainer error")
+    {
+    }
+
+    SoftwareContainerAgentError(const std::string &message):
+        m_message(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return m_message.c_str();
+    }
+
+protected:
+    std::string m_message;
+};
+
+class PreloadError : public SoftwareContainerAgentError
+{
+public:
+    PreloadError():
```

Comment:
I will remove this. Preloading will be removed, and any situation where this could have been used will soon be obsolete. Also I feel that in similar situations I could instead use the SCAgentError.